.github/workflows/update-snapshot.yml

```diff
@@ -0,0 +1,62 @@
+name: Update snapshots
+
+on:
+  # Run any time when a new label added to the PR 
+  pull_request:
+    types: [labeled]
+
+jobs:
+  update_snapshot:
+    name: Update snapshot by label
+    # This job will only run if the comment was on a pull requests and matches the label
+    if: ${{ github.event.label.name == 'ui change' }}
+    runs-on: ubuntu-latest
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE so job can access it
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          # Use a machine account when checking out. This is to workaround the issue were GitHub
+          # actions, when using the default account, cannot trigger other actions.
+          # This machine account is only for this PAT, pwd was created and thrown away
+          # If any update needed, create a new account, add access to the repo and generate a new PAT
+          token: ${{ secrets.MACHINE_ACCOUNT_PAT }}
+      # Ensure node version is great enough
+      - name: Use Node.js v14.x
+        uses: actions/setup-node@v1
+        with:
+          node-version: '14.x'
+      # Try get node_modules from cache
+      - name: Restore node_modules from cache
+        uses: actions/cache@v2
+        with:
+          path: common/temp/pnpm-store
+          key: ${{ runner.os }}-${{ hashFiles('common/config/rush/pnpm-lock.yaml') }}
+      # Install dependencies
+      - name: Install rush
+        run: npm install -g @microsoft/rush@5.47.0
+      - name: Install dependencies
+        run: rush install
+      - name: Build Packages
+        run: rush build -t @azure/communication-react
+      - name: Build
+        run: rush build -t sample-static-html-composites
+      - name: Update snapshot
+        run: |
+          cd samples/StaticHtmlComposites
+          rushx update:snapshot
+        env:
+          CONNECTION_STRING: ${{ secrets.CONNECTION_STRING }}
+      # Setup bot information for pushing new changes
+      # Here we use the id from the github actions bot: https://api.github.com/users/better-informatics%5Bbot%5D
+      - name: Setup bot git information
+        run: |
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config user.name "github-actions[bot]"
+      # Commit changes to snapshot files.
+      - name: Commit new snapshots
+        run: >
+          git diff --exit-code --quiet -- samples/StaticHtmlComposites/tests || 
+          (git add samples/StaticHtmlComposites/tests && 
+          git commit -m 'Updating automation snapshots' && 
+          git push origin HEAD:${{ github.head_ref }})
```

Comment:
@JamesBurnside Do you know how to reuse this part? Like create an external yml file and import it? This logic is reusable but not sure how to reuse it

Reply:
Not yet nicely that I've seen, its so sad 😔